### DISCUSSION TITLE
==> Build failed 😞

### DISCUSSION THREAD

**Discussion by Jason-Veloceo** [FlowiseAI/Flowise#1087] [General]
I am struggling to deploy Flowise on a Render instance. It runs for about 8 minutes, and then returns a Build Failed error.

flowise-ui:build: ERROR: command finished with error: command (/opt/render/project/src/packages/ui) yarn run build exited (1)
Oct 18 09:28:43 PM  command (/opt/render/project/src/packages/ui) yarn run build exited (1)
Oct 18 09:28:43 PM   ERROR  run failed: command  exited (1)
Oct 18 09:28:43 PM  error Command failed with exit code 1.
Oct 18 09:28:43 PM  info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Oct 18 09:28:43 PM  ==> Build failed 😞

It's doing this across several web services. I simply cannot get it to load. 

Does anyone know what this means? 

Thanks

**Comment by MASLER1987** [comment]
I'm also getting this error

**Reply by Addie909** [reply]
same here

**Reply by yumoxuan-25** [reply]
same

**Comment by AWillams2** [comment]
I have installed it locally today, can get it to open in localhost, create chatflows, but as soon as I try to message it returns 0: and the server stops with error Command failed with exit code 1.

yarn run v1.22.19
$ run-script-os
$ cd packages/server/bin && run start
2023-10-18 15:20:41 [INFO]: Starting Flowise...
2023-10-18 15:20:41 [INFO]: ⚡️ [server]: Flowise Server is listening at 3000
2023-10-18 15:20:41 [INFO]: 📦 [server]: Data Source has been initialized!
2023-10-18 15:21:13 [INFO]: ⬆️ POST /api/v1/chatmessage/ee615353-42cf-432e-9bfa-ff6cceba0238
2023-10-18 15:21:13 [INFO]: ⬆️ POST /api/v1/internal-prediction/ee615353-42cf-432e-9bfa-ff6cceba0238
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Reply by linkinleon** [reply]
Hey @AWillams2 I got the same issue before, then I found when I ran "yarn build", it seemed "success" in the end but there were some error messages in the middle like 'warning Error running install script for optional dependency: "/Users/XXX/project/saas-statistics/node_modules/canvas: Command failed', then I found this solution: https://github.com/Automattic/node-canvas/issues/2186#issuecomment-1369528914, after I brew installed all the packages and rebuilt the project, I don't have this issue any more.

**Reply by AWillams2** [reply]
Thanks @linkinleon, I ended up starting the clone/install/build process again today and that has now worked for me. Incidentally the issue seems to have been with some of the components (potentially csv upload, in-memory vector store) that were in the templates I tried and had issues with. When I did a simple flow (LLM/chain/prompt template) it worked.

**Comment by henri-edh** [comment]
I also get a BUILD failure on RENDER : flowise-ui:build:
Oct 18 08:01:14 PM  flowise-components:build: [18:01:14] Requiring external module ts-node/register
Oct 18 08:01:18 PM  flowise-components:build: [18:01:18] Using gulpfile ~/project/src/packages/components/gulpfile.ts
Oct 18 08:01:18 PM  flowise-components:build: [18:01:18] Starting 'default'...
Oct 18 08:01:18 PM  flowise-components:build: [18:01:18] Finished 'default' after 395 ms
Oct 18 08:03:31 PM  flowise-ui:build: Failed to compile.
Oct 18 08:03:31 PM  flowise-ui:build:
Oct 18 08:03:31 PM  flowise-ui:build: Cannot read properties of undefined (reading 'indexOf')
Oct 18 08:03:31 PM  flowise-ui:build:
Oct 18 08:03:31 PM  flowise-ui:build:
Oct 18 08:03:32 PM  flowise-ui:build: error Command failed with exit code 1.
Oct 18 08:03:32 PM  flowise-ui:build: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Oct 18 08:03:32 PM  
Oct 18 08:03:32 PM  flowise-ui:build: ERROR: command finished with error: command (/opt/render/project/src/packages/ui) yarn run build exited (1)
Oct 18 08:03:32 PM   Tasks:    1 successful, 2 total
Oct 18 08:03:32 PM  command (/opt/render/project/src/packages/ui) yarn run build exited (1)
Oct 18 08:03:32 PM  Cached:    0 cached, 2 total
Oct 18 08:03:32 PM    Time:    2m34.918s
Oct 18 08:03:32 PM  
Oct 18 08:03:32 PM   ERROR  run failed: command  exited (1)
Oct 18 08:03:32 PM  error Command failed with exit code 1.
Oct 18 08:03:32 PM  info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Oct 18 08:03:32 PM  ==> Build failed 😞

**Comment by HenryHengZJ** [comment]
related - #1094 

In short, simply redeploy latest commit, it should works.

I was having the same issue whole day, after spending whole day debugging, at the end of the day I simply redeploy latest commit without changing anything and it works. My guess is there was some underrlying dependencies that got updated and when you redeploy/fresh install, it pulls the latest and it works

**Reply by FlingwingDev** [reply]
Thanks Henry. It's still failing for me. I'm unsure what to do about it. I have tried instantiating a new web service - and it fails too. I'll try again later, but for now, it has me stumped. 

Cheers

**Reply by Addie909** [reply]
same here 
![image](https://github.com/FlowiseAI/Flowise/assets/148346367/26f76bbb-f38e-4d38-bf4a-f290b579d0a2)

**Reply by HenryHengZJ** [reply]
if you are still using the Node deployment method on Render, add this Env Variable: `NODE_VERSION`: `18.8.1`

**Comment by Jason-Veloceo** [comment]
So, it's not working for anyone at the moment? Has anyone been able to successfully deploy the commit. If not, how far back in commits do we need to go to get a stable version? I have gone 10 back, with the same issue.

**Reply by henri-edh** [reply]
same build issue for me still

**Reply by henri-edh** [reply]
I added to Environment Variables the following and rebuilt latest commit : `NODE_VERSION:`   ` lts`  This worked.
